### PR TITLE
Add post endpoint for Mux webhook

### PR DESCRIPTION
**Pull request by Ciszkoo**
`MUX_TOKEN_ID`, `MUX_TOKEN_SECRET` - needed for Mux client to upload video to Mux (will be in another branch)
`WEBHOOK_SECRET` - used in endpoint to verify upcoming events

We will be receiving events from Mux on this endpoint.

On `video.asset.ready` event we update upload's `assetId` column. (Have to double check if assetId is really needed)

On `video.asset.master.ready` event we update upload's `downloadUrl` column.

Mux Webhook requires `2xx` response status if event is received in order to stop producing this event. If it receive a different status it will be trying again with increasing timeout for the next 24 hours.

NOTE:
If possible update `videoEntries` table schema with TTL 24h for the `downoladUrl` column as master access url is valid for 24h.

### CODE REVIEW

File: src/server/db/schema.ts [hunk 1]
```diff
@@ -112,8 +112,9 @@ export const verificationTokens = mysqlTable(
 export const videoEntries = mysqlTable("videoEntry", {
   id: bigint("id", { mode: "number" }).primaryKey().autoincrement(),
   uploadId: varchar("uploadId", { length: 256 }).notNull(),
-  assetId: varchar("assetId", { length: 256 }),
+  assetId: varchar("assetId", { length: 255 }),
```
Comment (hubcio2115) [inline src/server/db/schema.ts:115]: Why you've changed length from `256` to `255`?

File: src/app/api/mux/route.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
+import Mux, { type Asset } from "@mux/mux-node";
```
Comment (hubcio2115) [inline src/app/api/mux/route.ts]: Only `Mux.Webhooks` maybe import just import `Webhooks`?

File: src/app/api/mux/route.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+export async function POST(req: NextRequest) {
+  try {
+    const signature = headers().get("mux-signature");
+
+    if (!signature) {
+      return new Response("Missing signature", { status: 400 });
+    }
+
+    const body = (await req.json()) as MuxWebhookAssetBody;
```
Comment (hubcio2115) [inline src/app/api/mux/route.ts]: You should instead create a zod schema that will `safeParse` the body.

**Review by hubcio2115 — CHANGES_REQUESTED** [review]

File: src/server/db/schema.ts (hunk 1 above)
Comment (Ciszkoo) [reply src/server/db/schema.ts:115]: Length of `assetId` is capped by mux to `255`.  Can't copy link to highlight for some reason: https://docs.mux.com/webhook-reference#video.asset.ready

File: src/app/api/mux/route.ts (hunk 2 above)
Comment (Ciszkoo) [reply src/app/api/mux/route.ts]: It looks like `Webhooks` needs an instance of `Mux` class and cannot be used imported (?)

File: src/app/api/mux/route.ts (hunk 3 above)
Comment (Ciszkoo) [reply src/app/api/mux/route.ts]: Schema added 👍

**Review by hubcio2115 — APPROVED** [review]
LGTM 👍🏻

**Pull request merged**